src/armTemplates/ase.ts

```diff
@@ -0,0 +1,82 @@
+import { FunctionAppResource } from "./resources/functionApp";
+import { AppInsightsResource } from "./resources/appInsights";
+import { StorageAccountResource } from "./resources/storageAccount";
+import { AppServicePlanResource } from "./resources/appServicePlan";
+import { HostingEnvironmentResource } from "./resources/hostingEnvironment";
+import { VirtualNetworkResource } from "./resources/virtualNetwork";
+import { ArmResourceTemplateGenerator, ArmResourceTemplate } from "../models/armTemplates";
+import { ServerlessAzureConfig } from "../models/serverless.js";
+
+const resources: ArmResourceTemplateGenerator[] = [
+  FunctionAppResource,
+  AppInsightsResource,
+  StorageAccountResource,
+  AppServicePlanResource,
+  HostingEnvironmentResource,
+  VirtualNetworkResource,
+];
+
+const AseTemplate: ArmResourceTemplateGenerator = {
+  getTemplate: () => {
+    const template: ArmResourceTemplate = {
+      "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+      "contentVersion": "1.0.0.0",
+      "parameters": {},
+      "resources": [],
+    };
+
+    resources.forEach((resource) => {
+      const resourceTemplate = resource.getTemplate();
+      template.parameters = {
+        ...template.parameters,
+        ...resourceTemplate.parameters,
+      };
+
+      template.resources = [
+        ...template.resources,
+        ...resourceTemplate.resources,
+      ];
+    });
+
+    template.parameters.appServicePlanSkuName.defaultValue = "I1";
+    template.parameters.appServicePlanSkuTier.defaultValue = "Isolated";
+
+    // Update the app service plan to point to the hosting environment
+    const appServicePlan: any = template.resources.find((resource: any) => resource.type === "Microsoft.Web/serverfarms");
```

Comment:
this method is huge.  I would extract this section and the one after into separate methods

Reply:
i have an idea to create a `CompositeArmTemplate` that abstracts all the common stuff away from these methods and will work on it.